qmi/core/transport.py

```diff
@@ -647,18 +630,38 @@ def read(self, nbytes: int, timeout: Optional[float]) -> bytes:
         while nbuf < nbytes:
             self._safe_socket.settimeout(tremain)
             try:
-                b = self._safe_socket.recv(nbytes - nbuf)
+                if self._assert_addr:
+                    # For UDP, reading data fails if the arriving packet size is larger than the given nbytes (- nbuf)
+                    # to read. To avoid this happening, read at least 4096 standard that is larger than regular MTU.
+                    udp_packet_size = max(nbytes - nbuf, 4096)
+                    try:
+                        b, addr = self._safe_socket.recvfrom(udp_packet_size)
+                    except (BlockingIOError, socket.timeout) as err:
+                        raise err  # Re-raise this for handling in the upper try-except.
+                    except OSError:
+                        raise QMI_RuntimeException(f"UDP packet size was larger than {udp_packet_size}. Data is lost.")
+
+                    if addr != self._address:
```

Comment:
Do you think this is even possible? In our current implementation. We'll only allow a single transport(socket), to be bound to one UDP device. So wouldn't we expect to only have packets from that address?

Reply:
It seemed to give some issues with testing with hardware and the unit-tests after some changes in the transport after hardware tests. So, it is indeed better to remove. In principle it COULD get some packets from elsewhere but I think only if they are sent to this specific port number opened for the UDP. Quite unlikely unless the PC has lots of UDP stuff going on for some reason. 

Removed all these checks.